agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -252,13 +192,13 @@ def start(info)
           '--password', ENV['KONTENA_TOKEN']
         ]
         exec_params += ['--trusted-subnets', trusted_subnets.join(',')] if trusted_subnets
-        self.exec(exec_params)
+        Actor[:weave_executor].execute(exec_params)
         weave = Docker::Container.get('weave') rescue nil
```

Comment:
I suspect this `Docker::Container.get('weave')` is broken with the async `launch-router` execute.

Particularly since the sleep-until loop below never re-gets it.